docs/UserGuide.md

```diff
@@ -195,6 +195,28 @@ Examples:
 
 * `lesson delete 2` deletes the 2nd lesson in the lessons schedule.
 
+### Finding lessons : `lesson find`
+
+Finds lessons whose student's name contain any of the given keywords.
+
+Format: `lesson find KEYWORD [MORE_KEYWORDS]`
+
+* The search is case-insensitive. e.g `hans` will match `Hans`
+* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* Only the name is searched.
+* Only full words will be matched e.g. `Han` will not match `Hans`
+* Lessons whose persons matching at least one keyword will be returned (i.e. `OR` search).
```

Comment:
Similar to previous comment, maybe you can try
"Lessons with students whose names match at least one of the specified keywords will be returned (i.e., an OR search)."

Reply:
Changed